hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/HollowScreening.java

```diff
@@ -0,0 +1,183 @@
+            if (correspondingKey != null) {
+                replacedPayerKey = correspondingKey;
+                pendingCompletions =
+                        maybeAddToCompletions(payerEvmAddress, correspondingKey, pendingCompletions, aliasManager);
+            }
+        }
+
+        List<JKey> replacedOtherKeys = null;
+        for (int i = 0; i < otherKeys.size(); i++) {
+            final var otherKey = otherKeys.get(i);
+            if (otherKey.hasWildcardECDSAKey()) {
+                final var hollowKey = otherKey.getWildcardECDSAKey();
+                final var evmAddress = hollowKey.getEvmAddress();
+                final var correspondingKey = evmAddressToEcdsaKeyIndex.get(Bytes.of(evmAddress));
+                if (correspondingKey != null) {
+                    if (replacedOtherKeys == null) {
+                        replacedOtherKeys = new ArrayList<>(otherKeys);
+                    }
+                    replacedOtherKeys.set(i, correspondingKey);
+                    if (hollowKey.isForHollowAccount()) {
+                        pendingCompletions =
+                                maybeAddToCompletions(evmAddress, correspondingKey, pendingCompletions, aliasManager);
+                    }
+                }
+            }
+        }
+        return new HollowScreenResult(pendingCompletions, replacedPayerKey, replacedOtherKeys);
+    }
+
+    /**
+     * Given a list of {@link TransactionSignature}, go through the signatures, and construct a
+     * look-up table for all present evm address <-> ECDSA key pairs in the sigs.
+     *
+     * <p>Serves as an efficient way of obtaining a {@link JECDSASecp256k1Key} corresponding to {@link JWildcardECDSAKey}
+     * in subsequent logic.
+     *
+     * @param txnSigs a list of {@link TransactionSignature}
+     * @return a {@code Map} of all present evm addresses <-> {@link JECDSASecp256k1Key} pairs in the {@code txnSigs}
+     */
+    private static Map<Bytes, JECDSASecp256k1Key> createEvmAddressToEcdsaKeyIndexFrom(
+            final List<TransactionSignature> txnSigs) {
+        final Map<Bytes, JECDSASecp256k1Key> evmAddressToEcdsaKeyIndex = new HashMap<>();
+        for (final var sig : txnSigs) {
+            if (sig.getSignatureType() != SignatureType.ECDSA_SECP256K1) {
+                continue;
+            }
+            final var expandedPublicKeyDirect = sig.getExpandedPublicKeyDirect();
+            final var evmAddress = MiscCryptoUtils.extractEvmAddressFromDecompressedECDSAKey(expandedPublicKeyDirect);
+            final var compressedSecp256k1 = MiscCryptoUtils.compressSecp256k1(expandedPublicKeyDirect);
+            final var key = new JECDSASecp256k1Key(compressedSecp256k1);
+
+            evmAddressToEcdsaKeyIndex.put(Bytes.of(evmAddress), key);
+        }
+        return evmAddressToEcdsaKeyIndex;
+    }
+
+    private static List<PendingCompletion> maybeAddToCompletions(
+            byte[] payerEvmAddress,
```

Comment:
This is used for both payer and other EVM addresses it seems, so maybe just `evmAddress` here instead of `payerEvmAddress`?

Reply:
Yess, renamed now